survival_problem_grad_descent.py

```diff
@@ -168,11 +168,13 @@ def _get_parallel_sum(worker_list, shared_obj):
         sorted_sample_labels = sorted(list(set(self.sample_labels)))
         log.info("Obtained gradients %s" % (time.time() - st))
 
-        # Get the expected scores
+        # Get the expected scores and their sum
         expected_scores = {label: 0 for label in sorted_sample_labels}
+        expected_scores_sum = 0
         for g, sample_label in zip(grad_log_lik, self.sample_labels):
             g = g.reshape((g.size, 1), order="F")
             expected_scores[sample_label] += g
+            expected_scores_sum += g
```

Comment:
I would move this expected sum calculation to a different loop. I think this would reduce computation time by a bit:

```
expected_scores_sum = 0
for sample_label in self.sample_labels:
    expected_scores_sum += expected_scores[sample_label]
```

Reply:
I can test this, too, but I don't see how it would be faster? Wouldn't it be looping over the same set of labels twice instead of once?

Reply:
I think it's faster cause you spent all that time calculating `expected_scores[sample_label]` and we dont need to repeat the same calculation for `expected_scores_sum`. We can just piggy back off of `expected_scores`.
It won't help by much, but maybe it will help a bit.